ext/puma_http11/org/jruby/puma/MiniSSL.java

```diff
@@ -196,6 +192,61 @@ public IRubyObject initialize(ThreadContext threadContext, IRubyObject miniSSLCo
     return this;
   }
 
+  /**
+   * Try to create a netty tcnative OpenSSL engine which has much better performance
+   * than the JDK native one.
+   * Fall back to the default engine otherwise.
+   */
+  private static SSLEngine tryCreateEngine(KeyManagerFactory keyManagerFactory,
+                                               TrustManagerFactory trustManagerFactory,
+                                               String[] protocols,
+                                               long verify_mode,
+                                               String[] ciphers) throws NoSuchAlgorithmException, KeyManagementException {
+    try {
+      io.netty.handler.ssl.ClientAuth clientAuth = io.netty.handler.ssl.ClientAuth.NONE;
+      if ((verify_mode & 0x1) != 0) {
+        clientAuth = io.netty.handler.ssl.ClientAuth.OPTIONAL;
+      }
+      if ((verify_mode & 0x2) != 0) {
```

Comment:
https://github.com/netty/netty/blob/2f32e0b8adb63decd9031e26fa5dd4154d93ce97/handler/src/main/java/io/netty/handler/ssl/JdkSslContext.java#L346
Are the associated mappings to verify.